# Hands-on Sesi 2

> Topik: Pandas & NumPy - Eksplorasi Dataset Titanic, Filtering, Groupby

Nama Lengkap : Rexxy Andrew Gunawan

NIM          : 240401010223

Kelas        : IF403

Program Studi: S1 PJJ Informatika UNSIA

Mata Kuliah  : Data Science

Disusun oleh : Rexxy Andrew Gunawan

Link Google Colab: https://colab.research.google.com/drive/10skpV2K1RGo9BEakc3dXCbkXEsFxypn3?usp=sharing

In [1]:
import pandas as pd
import numpy as np

# 1. Muat Dataset Titanic

In [2]:
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL) #karena ketika mengikuti link ppt error saya merubah link url untuk mengakses titanic.csv

# 2. Eksplorasi Awal

In [3]:
# periksa kolom dan tipe datanya (bisa dilihat juga apabila ada missing value)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [4]:
# periksa data statistik deskriptif
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [5]:
# memeriksa lebih lanjut kolom-kolom pada dataset serta missing value. (lebih mudah membaca missing value dengan 'df.isnull()')
print(f"Jumlah baris dan kolom: {df.shape}\n")
print(f"Kolom-kolom pada dataset: \n{df.columns.tolist()}\n")
print(f"Missing value pada dataset: \n {df.isnull().sum()}", 2*'\n', f"Missing value dalam persentase(%): \n {df.isnull().sum()/df.shape[0]*100}")

Jumlah baris dan kolom: (891, 12)

Kolom-kolom pada dataset: 
['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']

Missing value pada dataset: 
 PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64 

 Missing value dalam persentase(%): 
 PassengerId     0.000000
Survived        0.000000
Pclass          0.000000
Name            0.000000
Sex             0.000000
Age            19.865320
SibSp           0.000000
Parch           0.000000
Ticket          0.000000
Fare            0.000000
Cabin          77.104377
Embarked        0.224467
dtype: float64


## Q1
Berapa jumlah baris dan kolom dataseet Titanic? Kolom apa saja yang tersedia?

Jawab:

Pada dataset Titanic terdapat 891 baris dan 12 kolom.

Kolom-kolom yang ada pada dataset Titanic adalah :

1.   PassengerId
2.   Survived
3.   Pclass
4.   Name
5.   Sex
6.   Age
7.   SibSp
8.   Parch
9.   Ticket
10.   Fare
11.   Cabin
12.   Embarked

## Q2
Kolom mana saja yang memiliki missing values?
Berapa jumlah dan proporsinya (%)?

Jawab:

Kolom yang memiliki missing values adalah:


*   Age - 19.865320 %
*   Cabin - 77.104377 %
*   Embarked - 0.224467 %


# 3. Filtering

## Penumpang yang selamat

In [6]:
# Penumpang yang selamat adalah yang memiliki nilai df['Survived'] == 1
penumpang_selamat = df[df['Survived']==1]
print(f"Jumlah penumpang yang selamat : {penumpang_selamat.shape[0]} orang")
print(f"Persentase penumpang yang selamat : {penumpang_selamat.shape[0]/df.shape[0]*100} %")

Jumlah penumpang yang selamat : 342 orang
Persentase penumpang yang selamat : 38.38383838383838 %


## Q3
Berapa persen penumpang yang selamat (Survived=1)?

Jawab:

Persentase penumpang yang selamat adalah  38.38383838383838 %

##  Penumpang Wanita Kelas 1

In [7]:
wk1 = df[(df['Sex']=='female') & (df['Pclass']==1)] # filter penumpang wanita & di Kelas 1
print(f"Jumlah penumpang wanita di kelas 1 adalah : {wk1.shape[0]} orang")

wk1_survived = wk1[wk1['Survived']==1] # penumpang wanita di Kelas 1 'yang selamat'
print(f"Jumlah penumpang wanita yang selamat di Kelas 1 : {wk1_survived.shape[0]} orang")

print(f"Rata-rata penumpang wanita yang selamat di Kelas 1 : {wk1['Survived'].mean()*100} %")

print(f"Rata-rata usia penumpang kelas 1 adalah : {wk1['Age'].mean()} tahun")

print(f"Rata-rata usia penumpang wanita yang selamat di Kelas 1 adalah {wk1_survived['Age'].mean()} tahun")

Jumlah penumpang wanita di kelas 1 adalah : 94 orang
Jumlah penumpang wanita yang selamat di Kelas 1 : 91 orang
Rata-rata penumpang wanita yang selamat di Kelas 1 : 96.80851063829788 %
Rata-rata usia penumpang kelas 1 adalah : 34.61176470588235 tahun
Rata-rata usia penumpang wanita yang selamat di Kelas 1 adalah 34.9390243902439 tahun


## Q4

*    Rata-rata usia penumpang wanita kelas 1 yang selamat?
*   Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata
rata usianya?

Jawab:

*   Rata-rata usia penumpang wanita yang selamat di Kelas 1 adalah 34.9 tahun.
*   Jumlah penumpang wanita di kelas 1 adalah : 94 orang dan rata-rata usianya 34.6 tahun.



# 4. Groupby - Tingkat Keselamatan per Kelas


In [8]:
hasil = df.groupby('Pclass').agg(
  total=('PassengerId','count'),
  selamat=('Survived','mean'),
).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


## Q5
Gunakan groupby untuk menghitung tingkat keselamatan(mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?

Jawab:

Kelas yang memiliki tinggkat keselamatan paling tinggi adalah "Kelas 1".

In [9]:
# mencari variabel lain (jenis kelamin) - untuk Q6
hasil = df.groupby('Sex').agg(
  total=('PassengerId','count'),
  selamat=('Survived','mean'),
)
print(hasil)

        total   selamat
Sex                    
female    314  0.742038
male      577  0.188908


## Q6

Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik.
Sertakan interpretasi singkat.

Jawab:

Dilihat dari jenis kelamin yang selamat, penumpang wanita memiliki tingkat keselamatan yang jauh lebih tinggi dibandingkan penumpang laki-laki pada peristiwa tenggelamnya kapal Titanic. Hal ini kemungkinan ketika evakuasi penumpang wanita lebih diprioritaskan ketika evakuasi saat kejadian.

# Kesimpulan Sesi Pembelajaran Sesi 2

*   Apa yang dipelajari: Saya telah mempraktikkan penggunaan struktur data
Python (list, dictionary), serta dua library utama Data Science: NumPy untuk komputasi numerik dan Pandas untuk manipulasi data tabular (DataFrame).


*   Temuan utama: Melalui analisis dataset Titanic, ditemukan bahwa tingkat keselamatan sangat dipengaruhi oleh kelas penumpang (Kelas 1 paling tinggi yaitu 63%) dan jenis kelamin (penumpang wanita jauh lebih diprioritaskan untuk selamat dibanding laki-laki). Selain itu, ditemukan banyak data kosong (missing values) pada kolom 'Cabin' (77%) dan 'Age' (19%).


*   Keterbatasan/Pertanyaan: Saya masih perlu belajar cara yang tepat untuk menangani data yang hilang (missing values) tersebut agar tidak memengaruhi akurasi analisis di tahap selanjutnya.